### DISCUSSION TITLE
Include HTML comments for server side rendered Markup with out adding wrapper element

### DISCUSSION THREAD

**Discussion by RamAvuthu** [vercel/next.js#64613] [Ideas]
### Describe the feature you'd like to request

Need the option to include HTML comments in the server-side rendered markup.

I want to add `<!--googleoff: all-->` comment in the document for google on/off without adding the extra wrapper element.

### Describe the solution you'd like

I should be able to pass the comment to `dangerouslySetInnerHTML` using react fragment. So that the extra wrapper element will not be created.

### Describe alternatives you've considered

I tried with `renderToString` to convert the page markup as a string and then passed the string to `dangerouslySetInnerHTML` to render the markup. But the styles that I am injecting on each component(with `<style jsx>`) are missing.

**Comment by SukkaW** [comment]
I understand the feature you are requesting. However, implementing this on Next.js side would introduce extra performance overhead and it could break React DOM Stream SSR.
IMHO you can request the feature to `facebook/react` directly since it is related to how `ReactDOMServer` produces HTML.
